styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java

```diff
@@ -648,7 +649,9 @@ public void pollsPodStateWhenStarting() throws Exception {
     givenNextNaturalTrigger(HOURLY_WORKFLOW, "2016-03-14T16:00:00Z");
 
     givenActiveState(workflowInstance, RunState.create(workflowInstance,
-        RunState.State.RUNNING, StateData.newBuilder().trigger(TRIGGER1).build(),
+        RunState.State.RUNNING,
+        StateData.newBuilder().executionId("id").executionDescription(ExecutionDescription.forImage(
```

Comment:
There always needs to be an executionDescription in the State when the instance is in state SUBMITTING, SUBMITTED, and RUNNING.  Otherwise, we can't distinguish between flyte and docker workflows and which runner handler to run. https://github.com/spotify/styx/pull/835/files#diff-1c41a715741e2a5f27c1a47cae92c7a0R61

Reply:
Mmm, My understanding that those checks you just move it from [one place ](https://github.com/spotify/styx/pull/835/files#diff-1c41a715741e2a5f27c1a47cae92c7a0L116) to the [other](https://github.com/spotify/styx/pull/835/files#diff-1c41a715741e2a5f27c1a47cae92c7a0R73) then how it worked before?

Reply:
I think there were no check for executionDescription in state SUBMITTED or RUNNING. 

There is an exception for missing executionId in dockerRunner.poll(state), but this test injects a terminate event, so it doesn't fail without executionId. 

I could remove executionId as it doesn't matter in this case, but I thought maybe it would be good to send in a correct state. What do you think?